apps/gallery/test/marionette/activity_test.js

```diff
@@ -0,0 +1,156 @@
+/* global require, marionette, setup, __dirname */
+'use strict';
+
+var Gallery = require('./lib/gallery.js'),
+    GalleryActivityTester = require('./lib/galleryactivitytester.js'),
+    System = require('./lib/system.js'),
+    assert = require('assert');
+
+marionette('Gallery Activity Tests', function() {
+
+  var client, apps = {};
+
+  apps[GalleryActivityTester.ORIGIN] =
+    __dirname + '/apps/galleryactivitytester';
+
+  client = marionette.client({
+    profile: {
+      prefs: {
+        'device.storage.enabled': true,
+        'device.storage.testing': true,
+        'device.storage.prompt.testing': true
+      },
+      apps: apps
+    }
+  });
+
+  var galleryApp, activityTesterApp, system;
+  var imageInfo = {
+      name: 'pictures/firefoxOS.png',
+      type: 'image/png',
+      size: '52458'
+    };
+
+  setup(function() {
+    // Remove all files in temp device storage.
+    client.fileManager.removeAllFiles();
+    // Add file into the pictures directory
+    client.fileManager.add({
+      type: 'pictures',
+      filePath: 'test_media/Pictures/firefoxOS.png'
+    });
+    galleryApp = new Gallery(client);
+    activityTesterApp = GalleryActivityTester.create(client);
+    system = new System(client);
+  });
+
+  test('pick and open image using gallery pick and open activity', function() {
+    activityTesterApp.launch();
+    // Initiate pick activity by clicking pick image button
+    activityTesterApp.tapPickImageButton();
+
+    // Tap Gallery button from Systm app Action Menu
+    system.menuOptionButton('Gallery').tap();
+
+    // We've switched to the system app, and we're calling
+    // switchToApp to switch to gallery
+    system.switchToApp(Gallery.ORIGIN);
```

Comment:
I checked and switchToApp calls waitForApp inside it and throws error if it fails to switch. See
https://github.com/mozilla-b2g/gaia/blob/9dfedd7d35da00fa9b08dcffc3ab4f47da54e4f0/tests/jsmarionette/plugins/marionette-apps/lib/switchtoapp.js#L26

I will add comment to test explaining that switchToApp call waits. 

Also you are correct, since we have already switched to System app we don't need switchToFrame and can call directly client.apps.switchToApp(Gallery.ORIGIN) here.